test/correctness/cse_across_unroll_and_tuples.cpp

```diff
@@ -0,0 +1,54 @@
+#include <Halide.h>
+
+using namespace Halide;
+using namespace Halide::Internal;
+
+class CountSqrt : public IRMutator {
+    using IRMutator::visit;
+
+    Expr visit(const Call *op) override {
+        if (op->name == "sqrt_f32") {
+            result++;
+        }
+        return IRMutator::visit(op);
+    }
+
+public:
+    int result = 0;
+};
+
+int main(int argc, char **argv) {
+    // Check that values shared across tuple elements or unrolled loops only get
+    // computed once in the Halide IR. LLVM will hoist them if we don't, but
+    // compilation can be much faster if we do it earlier, especially if the
+    // unrolled loop or tuple is large.
+    for (int use_tuple = 0; use_tuple < 2; use_tuple++) {
```

Comment:
Nit: for this kind of thing, I personally prefer `for (bool use_tuple : {false, true})`

Reply:
done